src/socket.ts

```diff
@@ -319,7 +328,13 @@ export class ConstellationSocket extends EventEmitter {
     }
 
     private rebroadcastEvent(name: string) {
-        this.socket.addEventListener(name, evt => this.emit(name, evt));
+        const socket = this.socket;
+        socket.addEventListener(name, evt => {
+            // Sanity check that the socket is current.
+            if (this.socket === socket) {
```

Comment:
The problem was when reconnecting right after closing, the state check [here](https://github.com/WatchBeam/carina/blob/master/src/socket.ts#L177) would be checking too late as the state had already been set to Connecting inside of the connect call. This caused the socket to connect multiple times. Not sure of a nice way to handle this